src/main.js

```diff
@@ -0,0 +1,416 @@
+
+            //Add the include projects wrapper elements
+            Render.addParentProjectElements(selection.include_projects)
+
+            //Grab all projects by selection
+            doProjects(selection)
+
+            //Grab all Important BuildTypes
+            doImportant(selection)
+
+        })
+
+    })
+
+}
+
+async function doProjects(selection) {
+
+    //Iterate over give projects from selection and handle them asynchronous
+    for (let project of selection.include_projects) {
+        let parentProjectData = []
+        projectHandler(project, null, parentProjectData)
+    }
+}
+
+async function doImportant(selection) {
+
+    Render.updateQueue(true, 1)
+
+    // Create an important container in the project container style to append buildtypes to
+    let projectData = {}
+    projectData.important       = true
+    projectData.id              = 'importantBuildTypes'
+    projectData.name            = 'Important'
+    projectData.order           = 0
+    projectData.parentProjectId = 'important_wrapper'
+    Data.interpretProject(projectData)
+
+    // Iterate over all specified buildtypes in the important selection with consistent order
+    let order = 0
+    for (let buildTypeId of selection.important_buildtypes) {
+        order++
+        Render.updateQueue(true, 1)
+
+        // Query name data for containers from teamcity add some other specific data and send to handler.
+        let importantBuildType       = await Query.getImportantBuildType(buildTypeId)
+        importantBuildType.projectId = 'importantBuildTypes'
+        importantBuildType.order     = order
+        importantBuildType.suffix    = '_important'
+        buildTypeHandler(importantBuildType)
+
+        Render.updateQueue(false, 1)
+    }
+}
+
+async function projectHandler(project, order, parentProjectData) {
+
+    Render.updateQueue(true, 1)
+
+    //fetch data of the project with give Id
+    projectData = await Query.getProject(project)
+
+    projectData.order = order
+
+    parentProjectData.push({'id': project, 'testSuccess': 0, 'testTotal': 0})
+
+    Data.interpretProject(projectData)
+
+    //If there are buildTypes send them to handler
+    if (projectData.buildTypes.buildType) {
+        Object.entries(projectData.buildTypes.buildType).forEach(([key, buildType]) =>{
+            buildType.order  = key
+            buildType.suffix = '' 
+            buildTypeHandler(buildType, parentProjectData)
+        })
+    }
+
+    //If project has subprojects go through all
+    if (projectData.projects.project) {
+        //Place projects under buildTypes
+        let suborder = projectData.buildTypes?projectData.buildTypes.buildType.length:0
+
+        Object.entries(projectData.projects.project).forEach(([key, subProject]) => {
+            subProject.parentProjectData = [ ...parentProjectData ]
+            projectHandler(subProject.id, suborder+key, subProject.parentProjectData)
+        }) 
+    }
+
+}
+
+//Handle specified buildType
```

Comment:
Don't add comments by the function declaration. OInly if it is an complex algorithm. Instead try to make the function name more descriptive.